Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,25 +1,74 @@
+import React, { useState, FormEvent } from "react";
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
-  }
+  const [speed, setSpeed] = useState(NaN);
+  const [battery, setBattery] = useState(NaN);
+  const [weather, setWeather] = useState(0);
+  const [range, setRange] = useState(NaN);
+  const [speedError, setSpeedError] = useState("");
+  const [batteryError, setBatteryError] = useState("");
+  const calculateRange = (event: FormEvent) => {
+    event.preventDefault();
+    // speedError = "Speed and battery must be set.";
+    // Use the given equation to calculate the range
+    if (!Number.isNaN(speed) && !Number.isNaN(battery) && battery <= 100 && battery >=0 && speed <= 90 && speed >=0) {
+      const calculatedRange = -((speed * speed * battery) / 2500) + 4 * battery + weather;
```

Comment:
See above comment Do we need to store the range in a state variable? Can't we just derive it from the above 3 state variables?